test/test_helper.rb

```diff
@@ -6,3 +6,109 @@
 require 'flexmock/minitest'
 
 require 'pry'
+
+class TypeformDataTest < Minitest::Test
+
+  def typeform_mock_and_client_for(typeform_id, json)
+    client = TypeformData::Client.new(api_key: 'test-api-key')
+    typeform = client.typeform(typeform_id)
+
+    mock = flexmock(
+      'TypeformData::Client',
+      get: flexmock('Net::HTTP response', parsed_json: json)
+    )
+
+    [typeform, mock, client]
+  end
+
+end
+
+# Extend from this class when you want to run assertions against this library's handling of a
+# specific, mocked JSON response from Typeform containing responses to a particular Typeform.
+# See json_one_test.rb for an example.
+#
+# Subclasses should declare:
+#
+#   def test_object_graph
+#     assert_object_graph
+#     assert_that_our_api_key_is_not_serialized
+#   end
+#
+# To run assertions defined here. Given time constraints, I can't figure out right now how to set
+# up a test that doesn't run in this class, but runs in every subclass. Would love some help on
```

Comment:
Yeah! That definitely works. I think we should hold off on the metaprogramming for now though, to keep things simple / explicit. If that ends up being the best solution we should add it in later.